system/security/cryptohelper/ske.cpp

```diff
@@ -0,0 +1,161 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+/*
+ * Module dealing with openssl symmetric key encryption/decription.
+ * For now just AES.
+ */
+
+#if defined(_USE_OPENSSL) && !defined(_WIN32)
+
+#include "jliball.hpp"
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+#include "cryptocommon.hpp"
+#include "pke.hpp"
+#include "ske.hpp"
+
+namespace cryptohelper
+{
+
+size32_t aesKeyEncrypt(MemoryBuffer &out, size32_t inSz, const void *inBytes, const char key[aesKeySize], const char iv[aesBlockSize])
```

Comment:
What is the status of the aesEncrypt/aesDecrypt functions in jencrypt?  Are they superseded? How does the performance compare?  From a quick test it looks faster.  Should this code fall back to those implementations if no openssl, or should they be removed.

Reply:
I had forgotten they had existed!

>From a quick test it looks faster.
did you mean the jlib versions are faster?

>Should this code fall back to those implementations if no openssl, or should they be removed.
well it's annoying that openssl is conditional..
But feels like we should consolidate to one implementation.

Reply:
@ghalliday - will do so as part of HPCC-20426